src/moments.jl

```diff
@@ -363,3 +363,44 @@ end
 
 kurtosis(v::RealArray) = kurtosis(v, mean(v))
 kurtosis(v::RealArray, wv::AbstractWeights) = kurtosis(v, wv, mean(v, wv))
+
+"""
+    cumulant(v, k, [wv::AbstractWeights], m=mean(v))
+
+Return the `k`th order cumulant of a real-valued array `v`, optionally
+specifying a weighting vector `wv` and a center `m`. This quantity is calculated using a recursive definition on lower order cumulants and central moments.
+"""
+function cumulant(v::RealArray, k::Int, m::Real)
+    cmoms = zeros(Float64, k)
+    cumls = zeros(Float64, k)
+    cmoms[1] = 0
+    cumls[1] = m
+    for i=2:k
+        @inbounds cmoms[i] =  moment(v, i, m)
+        @inbounds kn = cmoms[i]
+        for j=2:i-2
+            @inbounds kn -= binomial(i-1, j)*cmoms[j]*cumls[i-j]
+        end
+        @inbounds cumls[i] = kn
+    end
```

Comment:
`@inbounds` is probably not worth it here given that most of the work will happen in the call to `moment`:
```suggestion
    for i = 2:k
        kn = moment(v, i, m)
        for j = 2:i-2
             kn -= binomial(i-1, j)*cmoms[j]*cumls[i-j]
        end
        cumls[i] = kn
    end
```